src/main/java/com/ably/kafka/connect/ChannelSinkTask.java

```diff
@@ -47,25 +58,38 @@ public void start(Map<String, String> settings) {
         } catch (ChannelSinkConnectorConfig.ConfigException e) {
             logger.error("Failed to create Ably client", e);
         }
-        ablyClient.connect(isSuspended -> {
-            synchronized(ChannelSinkTask.this){
-                suspended.set(isSuspended);
-                if (!isSuspended){
-                    processSuspendQueue();
-                }
-            }
-        });
+//        ablyClient.connect(isSuspended -> {
+//            synchronized(ChannelSinkTask.this){
+//                suspended.set(isSuspended);
+//                if (!isSuspended){
+//                    processSuspendQueue();
+//                }
+//            }
+//        });
+        logger.info("Starting Ably channel Sink task");
+        // start the Batch processing thread.
+        this.sinkRecords = new ConcurrentLinkedQueue<>();
+        this.batchProcessingThread = new BatchProcessingThread(this.sinkRecords, this.ablyClient);
+        this.executor = new BatchProcessingExecutor(Integer.parseInt(settings
+                .get(ChannelSinkConnectorConfig.BATCH_EXECUTION_THREAD_POOL_SIZE)));
+
+        this.executor.scheduleAtFixedRate(this.batchProcessingThread, 0,
+                Integer.parseInt(settings.get(ChannelSinkConnectorConfig.BATCH_EXECUTION_FLUSH_TIME)),
+                TimeUnit.MILLISECONDS);
+
     }
 
     @Override
     public void put(Collection<SinkRecord> records) {
         if (ablyClient == null) {
-            throw new ConnectException("Ably client is unitialized");
+            throw new ConnectException("Ably client is uninitialized");
         }
 
-        for (final SinkRecord record : records) {
-            publishSingleRecord(record);
-        }
+        this.sinkRecords.addAll(records);
```

Comment:
Is there any throttling or otherwise limits on how frequently `put()` is called by the framework? It looks to me like this is now effectively quite a tight loop putting records directly from the topic partition into an unbounded queue?

When do records ever get removed from this queue? Presumably it should be after farming them off to a worker thread, or successful transmission to Ably? Seems like there's a race here where the queue could be come extremely large if there's a lot of data waiting in the topic partition (e.g. on startup, after redeploy or recovery), depending on how frequently the `BatchProcessingThread` is configured to run?